pkg/kubelet/kubelet_pods.go

```diff
@@ -2130,6 +2145,39 @@ func (kl *Kubelet) convertToAPIContainerStatuses(pod *v1.Pod, podStatus *kubecon
 		if oldStatus != nil {
 			status.VolumeMounts = oldStatus.VolumeMounts // immutable
 		}
+		if utilfeature.DefaultFeatureGate.Enabled(features.ImageVolume) && utilfeature.DefaultFeatureGate.Enabled(features.ImageVolumeWithDigest) && imageVolumeNames.Len() > 0 {
+			for i := range status.VolumeMounts {
+				volumeName := status.VolumeMounts[i].Name
+				if !imageVolumeNames.Has(volumeName) {
+					continue
+				}
+
+				var imageSpec kubecontainer.ImageSpec
+				for _, curVolumeMount := range cs.Mounts {
+					if curVolumeMount.Image == nil || curVolumeMount.Name != volumeName {
+						continue
+					}
+					imageSpec = kuberuntime.ToKubeContainerImageSpec(&runtimeapi.Image{
+						Id:   curVolumeMount.Image.Image,
+						Spec: curVolumeMount.Image,
+					})
+					break
+				}
+
+				if imageSpec.Image == "" {
+					klog.ErrorS(fmt.Errorf("image was not found"), "", "pod", pod.Name, "container", cs.Name, "volume name", volumeName)
+					continue
+				}
+
+				imageRef, err := kl.containerRuntime.GetImageRef(context.Background(), imageSpec)
```

Comment:
if you want to validate some of this logic, especially error cases, it can be done via criproxy tests